packages/accounts/src/light-account/account.ts

```diff
@@ -188,7 +229,21 @@ export class LightSmartContractAccount<
       data,
     });
 
-    provider.account.owner = newOwner;
+    const accountAddress = await provider.getAddress();
+    const initCode = await provider.account.getInitCode();
+    provider.connect(
+      (rpcClient) =>
+        new LightSmartContractAccount({
+          rpcClient,
+          chain: rpcClient.chain,
+          owner: newOwner,
+          entryPointAddress: provider.account.getEntryPointAddress(),
+          factoryAddress: provider.account.getFactoryAddress(),
+          index: provider.account.index,
+          initCode,
+          accountAddress,
```

Comment:
is there a reason reconnect the provider with the same initCode as the original provider's account? can't we just set the owner to the new owner?

also, do we need to first disconnect at all if we're going to reconnect?

Reply:
yea I don't know if we do need it since we send a UO anyways so it will get deployed with the only account, but just in case I wanted to keep it here

Reply:
I was disconnecting first since I wanna make some changes to disconnect to clear out the provider methods that connect adds